roxie/roxiemem/roxiemem.cpp

```diff
@@ -1695,7 +1777,7 @@ class FixedSizeHeaplet : public ChunkedHeaplet
             const char *block = data() + base;
             if (heapFlags & RHFnofragment)
             {
-                if (((std::atomic_uint *)block)->load(std::memory_order_relaxed) != FREE_ROW_COUNT)
+                if (((std::atomic_uint *)block)->load(std::memory_order_relaxed) < FREE_ROW_COUNT_MIN)
```

Comment:
ok, rethinking some more.  "all previous" should have been "all subsequent"

In general decrement/store to the link count should be an release operation, and any read should be an acquire - so that any pending write operations will consistent.  However there are some restrictions in the roxie mem usage:
- all allocations are protected by an acquire/release spinblock (or cs).
- Only the thread that allocates the memory modifies it.  As long as rows are passed to other threads with acquire/release this means that the row will always be up to date in all threads.
Therefore decrements do not need release (since memory is already synchronized, schronized by the alloation cs, or not shared with another thread), and accesses do not need to use acquire.
- Do you agree with the logic?  If so it should be added as an explicit comment to the head of the file.